mountpoint-s3-client/tests/get_object.rs

```diff
@@ -338,3 +339,79 @@ async fn test_get_object_cancel(read: bool) {
     // were actually cancelled, but we can at least check that the drop doesn't panic/deadlock.
     drop(request);
 }
+
+#[test_case(1, HashMap::from([("foo".to_string(), "bar".to_string())]); "1-byte object with metadata")]
+#[test_case(10, HashMap::from([("foo".to_string(), "bar".to_string())]); "small object with metadata")]
+#[test_case(30000000, HashMap::from([("foo".to_string(), "bar".to_string())]); "large object with metadata")]
+#[tokio::test]
+async fn test_get_object_user_metadata(size: usize, metadata: HashMap<String, String>) {
+    let sdk_client = get_test_sdk_client().await;
+    let (bucket, prefix) = get_test_bucket_and_prefix("test_get_object_user_metadata");
+
+    let key = format!("{prefix}/test");
+    let body = vec![0x42; size];
+    sdk_client
+        .put_object()
+        .bucket(&bucket)
+        .key(&key)
+        .set_metadata(Some(metadata.clone()))
+        .body(ByteStream::from(body.clone()))
+        .send()
+        .await
+        .unwrap();
+
+    let client: S3CrtClient = get_test_client();
+
+    let result = client
+        .get_object(&bucket, &key, None, None)
+        .await
+        .expect("get_object should succeed");
+    pin_mut!(result);
```

Comment:
Should we test that we can also call `get_object_metadata()` **after** streaming the body?

Reply:
I'd also like to see a test case for a scenario when the meta request fails with an error, for a non-existing object for example. Will a `S3RequestError::RequestCanceled` be returned for `get_object_metadata`?